src/lib/react/macro/ReactTypeMacro.hx

```diff
@@ -106,93 +106,18 @@ class ReactTypeMacro
 		propsType:ComplexType,
 		stateType:ComplexType
 	) {
-		var partialStateType = switch (ComplexTypeTools.toType(stateType)) {
-			case TType(_):
-				TPath({
-					name: 'Partial',
-					pack: ['react'],
-					params: [TPType(stateType)]
-				});
-
-			default:
-				macro :Dynamic;
-		};
-
-		var setStateArgs:Array<FunctionArg> = [
-			{
-				name: 'nextState',
-				// TState -> Partial<TState>
-				type: TFunction([stateType], partialStateType),
-				opt: false
-			},
-			{
-				name: 'callback',
-				type: macro :Void->Void,
-				opt: true
-			}
-		];
-
-		fields.push({
-			name: 'setState',
-			access: [APublic, AOverride],
-			meta: [
-				{
-					// Add @:extern meta so that this code only exist at compile time
-					name: ':extern',
-					params: null,
-					pos: Context.currentPos()
-				},
-				{
-					// First overload:
-					// function(nextState:TState -> TProps -> Partial<TState>, ?callback:Void -> Void):Void {}
-					name: ':overload',
-					params: [generateSetStateOverload(
-						TFunction([stateType, propsType], partialStateType)
-					)],
-					pos: Context.currentPos()
-				},
-				{
-					// Second overload:
-					// function(nextState:Partial<TState>, ?callback:Void -> Void):Void {}
-					name: ':overload',
-					params: [generateSetStateOverload(partialStateType)],
-					pos: Context.currentPos()
-				}
-			],
-			kind: FFun({
-				args: setStateArgs,
-				ret: macro :Void,
-				#if haxe4
-				expr: null
+		fields.push((macro class C {
+			@:extern
+			@:overload(function(nextState:$stateType -> react.Partial<$stateType>, ?callback:Void -> Void):Void {})
+			@:overload(function(nextState:$stateType -> $propsType -> react.Partial<$stateType>, ?callback:Void -> Void):Void {})
+			override public function setState(state: react.Partial<$stateType>, ?callback:Void -> Void): Void
```

Comment:
I'd prefer `nextState` instead of `state` here as well to avoid confusion (both regarding overloaded signatures & function body mismatch, and confusion with the `state` class field).